source/common/websocket/codec.cc

```diff
@@ -0,0 +1,219 @@
+#include "source/common/websocket/codec.h"
+
+#include <algorithm>
+#include <array>
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+#include "source/common/buffer/buffer_impl.h"
+#include "source/common/common/scalar_to_byte_vector.h"
+
+namespace Envoy {
+namespace WebSocket {
+
+Encoder::Encoder() = default;
+
+void Encoder::newFrameHeader(const Frame& frame, std::vector<uint8_t>& output) {
```

Comment:
nit: Instead of returning void and taking output as an outparam, let's return the output vector.